pkg/postgres/pgutils/error.go

```diff
@@ -0,0 +1,58 @@
+package pgutils
+
+import (
+	"io"
+	"net"
+
+	"github.com/jackc/pgconn"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+var transientPGCodes = set.NewFrozenStringSet(
+	// Class 08 — Connection Exception
+	"08000", // connection_exception
+	"08003", // connection_does_not_exist
+	"08006", // connection_failure
+	"08001", // sqlclient_unable_to_establish_sqlconnection
+	"08004", // sqlserver_rejected_establishment_of_sqlconnection
+	"08007", // transaction_resolution_unknown
+	"08P01", // protocol_violation
+
+	// Class 40 — Transaction Rollback
+	"40000", // transaction_rollback
+	"40002", // transaction_integrity_constraint_violation
+	"40001", // serialization_failure
```

Comment:
How do you tell which codes are transient? Like this one and database_dropped, it does not look like they can be corrected by retries.

Reply:
I was reading some notes on the Postgres page. I think we could remove the dropped database one

Reply:
Both Repeatable Read and Serializable isolation levels can produce errors that are designed to prevent serialization anomalies. As previously stated, applications using these levels must be prepared to retry transactions that fail due to serialization errors. Such an error's message text will vary according to the precise circumstances, but it will always have the SQLSTATE code 40001 (serialization_failure).